crypto/http/http_client.c

```diff
@@ -285,10 +286,14 @@ static int set_content(OSSL_HTTP_REQ_CTX *rctx,
             && BIO_printf(rctx->mem, "Content-Type: %s\r\n", content_type) <= 0)
         return 0;
 
-    rctx->req = req;
-    if ((req_len = BIO_ctrl(req, BIO_CTRL_INFO, 0, NULL)) <= 0)
-        return 1; /* streaming BIO may not support querying size */
-    return BIO_printf(rctx->mem, "Content-Length: %ld\r\n", req_len) > 0;
+    /* streaming BIO may not support querying size */
+    if ((req_len = BIO_ctrl(req, BIO_CTRL_INFO, 0, NULL)) <= 0
+        || BIO_printf(rctx->mem, "Content-Length: %ld\r\n", req_len) > 0) {
+        BIO_free(rctx->req);
```

Comment:
Good point to add this as a caveat (though `rctx->req` should be NULL before).

Reply:
Even with keepalive?

Reply:
Indeed, there it is likely needed :+1:

Reply:
When re-using `rctx` (typically with keep-alive), I just noticed that we need also a way of clearing `rctx->req` in case the follow-up request is a GET (i.e, has an empty req body),
but I'll take care of that in a follow-up PR (along with renaming `OSSL_HTTP_set_request`).